pkg/network/protocols/http/statkeeper.go

```diff
@@ -160,6 +160,16 @@ func (h *StatKeeper) add(tx Transaction) {
 		return
 	}
 
+	// Validate HTTP status code
+	statusCode := tx.StatusCode()
+	if statusCode < 100 || statusCode > 599 {
+		h.telemetry.invalidStatusCode.Add(1)
+		if h.oversizedLogLimit.ShouldLog() {
+			log.Warnf("invalid status code: %s", tx.String())
+		}
+		return
+	}
```

Comment:
That already exists in AddRequest - https://github.com/DataDog/datadog-agent/blob/abb5cfcf3a7e368e1313ba1a48f9f3e35813e2ae/pkg/network/protocols/http/stats.go#L208-L210

Do we need to move the original condition to here? Is there a good reason for moving it?